### DISCUSSION TITLE
PCM1808 ADC produces loud white noise through bluetooth

### DISCUSSION THREAD

**Discussion by Midnight-Robotics** [pschatzmann/arduino-audio-tools#1130] [Q&A] [answered]
Hello!

I'm trying to use the library to create a bluetooth source that can be used to interface my Sony Walkman with bluetooth headphones. I chose to use [this]https://www.amazon.com/PCM1808-Single-Ended-Analog-Input-Decoder-Amplifier/dp/B0925CBM5T/ref=sr_1_2?crid=3AIP4EA0I9ZGS&keywords=pcm1808&qid=1703897724&s=electronics&sprefix=pcm1808%2Celectronics%2C116&sr=1-2l) PCM1808 module to interface a 3.5mm jack with I2S. The bluetooth seems to work fine, but I get this nasty loud white noise in this rhythmic pattern for the whole time the ESP32 is powered on. I hooked it up exactly as the last part of the "external ADC" article in the wiki said. My code is as follows:
`#include "AudioTools.h"
#include "AudioLibs/AudioA2DP.h"

I2SStream i2sStream;                            // Access I2S as stream
A2DPStream a2dpStream;                          // access A2DP as stream
StreamCopy copier(a2dpStream, i2sStream); // copy i2sStream to a2dpStream

// Arduino Setup
void setup(void) {
    Serial.begin(115200);
    AudioLogger::instance().begin(Serial, AudioLogger::Info);
    
    // start bluetooth
    Serial.println("starting A2DP...");
    auto cfgA2DP = a2dpStream.defaultConfig(TX_MODE);
    cfgA2DP.name = "WH-1000XM4";
    a2dpStream.begin(cfgA2DP);

    // set intial volume
    a2dpStream.setVolume(0.3);

    // start i2s input with default configuration
    Serial.println("starting I2S...");
    auto cfg = i2sStream.defaultConfig(RX_MODE);
    cfg.i2s_format = I2S_STD_FORMAT; // or try with I2S_LSB_FORMAT
    cfg.bits_per_sample = 32;
    cfg.channels = 2; // optional because default setting
    cfg.sample_rate = 44100;
    cfg.is_master = true; // optional because default setting
    // this module needs a master clock if the ESP32 is master
    cfg.use_apll = false;  
    cfg.pin_mck = 3;  // i2s is using the info from a2dp
    i2sStream.begin(i2sStream.defaultConfig(RX_MODE));

}

// Arduino loop - copy data
void loop() {
   // copier.copy(filler);
    copier.copy();
}`

**Comment by joba-1** [comment]
you don’t use cfg in the last setup line?

**Comment by pschatzmann** [comment, marked as answer]
You completely messed up with the format: if you copy the input to the output format needs to match!
In addition A2DP only supports the format that you can find in the documentation.

So you need to find out if your PCM1808 suppports 16 bits and if not you need to change the format dynamically as described in the Wiki.
As joba-1 already outlined: the last line or the lines before don't make any sense and you need to use one or the other...

The Wiki already describes how to [test the ADC seperatly](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working)

**Reply by Midnight-Robotics** [reply]
thank you. I'll begin looking into this soon. I think I accidentally hit undo while copying and the last line reverted, but I did use the cfg I set up for the adc. Thanks for the pointers!
